euphoria-core/src/main/java/cz/seznam/euphoria/core/client/operator/ReduceByKey.java

```diff
@@ -352,19 +391,58 @@ public boolean isCombinable() {
     return DAG.of(reduceState);
   }
 
+  static <VALUE> ReduceFunctor<VALUE, VALUE> toReduceFunctor(
+      CombinableReduceFunction<VALUE> reducer1) {
+
+    return new ReduceFunctor<VALUE, VALUE>() {
+      @Override
+      public boolean isCombinable() {
+        return true;
+      }
+
+      @Override
+      public void apply(Iterable<VALUE> elem, Context<VALUE> context) {
+        context.collect(reducer1.apply(elem));
+      }
+    };
+  }
+
+
+  public static class SingleValueContext<T> implements Context<T> {
+
+    T value;
+
+    @Override
+    public void collect(T elem) {
+      value = elem;
+    }
+
+    @Override
+    public Object getWindow() {
+      throw new UnsupportedOperationException(
+          "The window is unknown when applying combinable function");
```

Comment:
I just realized the original problem of combinable `ReduceFunctor` to have access to the reduced window - if the windowing is merging windowing, then the window label will change during reducing (i.e. some elements will be reduced as if the window was X, while some other would be reduced as if the window would be Y, and the results will be randomly mixed together). I think it is better not to allow user code to access the window by throwing the `UnsupportedOperationException`. We could make this dependent on the merging windowing, but I think that a more straightforward approach is just to disable it all at once to stick to the least astonishment principle.